lib/tools/adb-commands.js

```diff
@@ -707,8 +707,24 @@ methods.getDeviceProperty = async function (property) {
 };
 
 methods.setDeviceProperty = async function (prop, val) {
+  let apiLevel = await this.getApiLevel(), err;
+  if (apiLevel >= 26 ) {
+    log.debug(`Running adb root, Android O needs adb to be rooted to setDeviceProperty`);
+    await this.root();
+  }
   log.debug(`Setting device property '${prop}' to '${val}'`);
-  await this.shell(['setprop', prop, val]);
+  try {
+    await this.shell(['setprop', prop, val]);
+  } catch (e) {
+    err = e;
+  }
+  if (apiLevel >= 26 ) {
+    log.debug(`Removing adb root for setDeviceProperty`);
+    await this.unroot();
+  }
+  if (!_.isUndefined(err)) {
+    log.errorAndThrow(`Error trying to set device property. Original error: ${err.message}`);
```

Comment:
throwing the error here via `log.errorAndThrow` will show you the current function as the the stacktrace target, but the actual exception happens inside `this.shell` call. So it will be complicated to find the real problem source. On the other side `throw` preserves the original stracktrace, so it will be visible where the error actually happened